java/src/hu/holyoil/controller/GameController.java

```diff
@@ -182,6 +193,184 @@ public void CheckGameCondition()  {
     public void StartGame()  {
         Logger.Log(this,"Starting game");
         // todo
+        // Generate between 100 and 500 asteroids
+        Random random = new Random();
+        int numOfAsteroids = 100;
+        if (Main.isRandomEnabled) {
+            numOfAsteroids = random.nextInt(401) + 100;
+        }
+
+        String startingAsteroidName = null;
+        String ufoStartingAsteroidName = null;
+
+        for (int i = 0; i < numOfAsteroids; i++) {
+
+            Asteroid asteroid = new Asteroid(
+                    AbstractBaseRepository.GetIdWithPrefix("Asteroid")
+            );
+
+            if (startingAsteroidName == null)
+                startingAsteroidName = asteroid.GetId();
+
+            ufoStartingAsteroidName = asteroid.GetId();
+
+            asteroid.SetNumOfLayersRemaining(
+                    Main.isRandomEnabled ? random.nextInt(6) : (i % 6)
+            );
```

Comment:
This method doesn't need to handle `Math.isRandomEnabled`, as it is purely for playing purposes. Not testing.

Reply:
Should all asteroids start with a layer depth of 5?

Reply:
Hmmm maybe 4-6?